### PR TITLE
Fix the queue name problem. Support multiple queues

### PR DESCRIPTION
**Pull request by sunloverz**

### CODE REVIEW

**Comment by rkononov** [comment]
@thousandsofthem could you take a look

File: README.md [hunk 1]
```diff
@@ -36,16 +36,15 @@ gem 'delayed_job_ironmq', '2.0.0.pre1'
...
-Delayed::Worker.configure do |config|
-  # optional params:
-  config.available_priorities = [-1,0,1,2] # Default is [0]. Please note, adding new priorities will slow down picking the next job from queue.  Also note that these priorities must include all priorities of your Delayed Jobs.
-  config.queue_name = 'default' # Specify an alternative queue name
-  config.delay = 0  # Time to wait before message will be available on the queue
-  config.timeout = 5.minutes # The time in seconds to wait after message is taken off the queue, before it is put back on. Delete before :timeout to ensure it does not go back on the queue.
-  config.expires_in = 7.days # After this time, message will be automatically removed from the queue.
-  config.error_queue = 'error_queue' # The failed jobs will be placed into the error queue
-end
+Delayed::IronMqBackend..available_priorities = [-1,0,1,2] # Default is [0]. Please note, adding new priorities will slow down picking the next job from queue.  Also note that these priorities must include all priorities of your Delayed Jobs.
```
Comment (thousandsofthem) [inline README.md]: Double dots?

File: README.md [hunk 2]
```diff
@@ -36,16 +36,15 @@ gem 'delayed_job_ironmq', '2.0.0.pre1'
...
+Delayed::IronMqBackend..queue_name = 'default' # Specify a default queue name
+Delayed::IronMqBackend..delay = 0  # Time to wait before message will be available on the queue
+Delayed::IronMqBackend..timeout = 5.minutes # The time in seconds to wait after message is taken off the queue, before it is put back on. Delete before :timeout to ensure it does not go back on the queue.
+Delayed::IronMqBackend..expires_in = 7.days # After this time, message will be automatically removed from the queue.
+Delayed::IronMqBackend..error_queue = 'error_queue' # The failed jobs will be placed into the error queue
+Delayed::IronMqBackend.queues = ['first', 'second', 'third']
```
Comment (thousandsofthem) [inline README.md]: could you add description and also describe how it's different from `.queue_name` ?

File: lib/delayed/backend/actions.rb [hunk 3]
```diff
@@ -30,31 +30,37 @@ def db_time_now
...
+              return [Delayed::Backend::Ironmq::Job.new(message)] if message
             end
-            return [Delayed::Backend::Ironmq::Job.new(message)] if message
           end
           []
         end
 
         def delete_all
-          Delayed::Worker.available_priorities.each do |priority|
+          Delayed::IronMqBackend.available_priorities.each do |priority|
```
Comment (thousandsofthem) [inline lib/delayed/backend/actions.rb:49]: found it. all good

File: lib/delayed/backend/actions.rb [hunk 4]
```diff
@@ -30,31 +30,37 @@ def db_time_now
...
             loop do
               msgs = nil
-              queue = queue_name(priority)
-              begin
-                msgs = ironmq.queue(queue).get(:n => 1000)
-              rescue Exception => e
-                Delayed::Worker.logger.warn(e.message)
+              Delayed::IronMqBackend.queues.each do |queue_item|
```
Comment (thousandsofthem) [inline lib/delayed/backend/actions.rb:52]: found it. all good

File: lib/delayed/backend/actions.rb [hunk 5]
```diff
@@ -30,31 +30,37 @@ def db_time_now
...
+                queue = queue_name(queue_item, priority)
+                begin
+                  msgs = ironmq.queue(queue).get(:n => 1000)
```
Comment (thousandsofthem) [inline lib/delayed/backend/actions.rb]: while you're around: n=1000 is not supported by IronMQ

File: lib/delayed/backend/iron_mq_backend.rb [hunk 6]
```diff
@@ -1,18 +1,18 @@
 require_relative 'iron_mq_config'
 
 module Delayed
-  class Worker
+  class IronMqBackend
     class << self
       attr_accessor :config, :ironmq, :queue_name, :delay, :timeout, :expires_in,
```
Comment (thousandsofthem) [inline lib/delayed/backend/iron_mq_backend.rb]: what `queue_name` exactly is and how you're using it?

File: README.md (hunk 2 above)
Comment (sunloverz) [inline README.md]: > Delayed::IronMqBackend.queues = ['first', 'second', 'third'] 

To support multiple queues `job.perform(queue: 'first')` or `job.perfrom(queue: second)`. Before it was only default queue

File: lib/delayed/backend/iron_mq_backend.rb (hunk 6 above)
Comment (sunloverz) [inline lib/delayed/backend/iron_mq_backend.rb]: you mean the following?

```
def queue_name(queue, priority)
   "#{queue}_#{priority || 0}"
end
```

Comment (thousandsofthem) [inline lib/delayed/backend/iron_mq_backend.rb]: I mean i see number of various uses, e.g. `attr_accessor : queue_name` means something different than 

```
def queue_name(queue, priority)
   "#{queue}_#{priority || 0}"
end
```

which one is true and how it supposed to work?

**Comment by thousandsofthem** [comment]
also, readme still confuzing. This means all the jobs by default going to `.queue_name` and someone can  define queue name from `.queues` list. right? also, `default` is not in the `[..]` list

```
Delayed::IronMqBackend.queue_name = 'default' # Specify a default queue name
Delayed::IronMqBackend.queues = ['first', 'second', 'third'] # The queues that you want your job to run in
```

File: lib/delayed/backend/iron_mq_backend.rb (hunk 6 above)
Comment (sunloverz) [inline lib/delayed/backend/iron_mq_backend.rb]: `attr_accessor : queue_name` is a default queue, `default_queue` seems better name

**Comment by sunloverz** [comment]
> also, readme still confuzing. This means all the jobs by default going to .queue_name and someone can define queue name from .queues list. right? also, default is not in the [..] list

Yes,a user should specify all queues first. I'll rename `.queue_name` to `default_queue` and add it to the queues list

File: lib/delayed/backend/iron_mq_backend.rb (hunk 6 above)
Comment (sunloverz) [inline lib/delayed/backend/iron_mq_backend.rb]: Before it was only one default queue for all tasks, now `task.perform_later(queue: 'test_queue')` will run task in `test_queue` and user should specify this queue in the queue list `Delayed::IronMqBackend.queues = ['test_queue', ...]`

**Comment by sunloverz** [comment]
@thousandsofthem could you have a look?

**Comment by sunloverz** [comment]
@thousandsofthem Can we merge this PR?

**Comment by thousandsofthem** [comment]
@sunloverz lgtm, can merge

**Comment by sunloverz** [comment]
Thanks!

**Pull request merged**